pkg/rancher-desktop/config/settings.ts

```diff
@@ -317,39 +352,78 @@ export function updateFromCommandLine(cfg: Settings, commandLineArgs: string[]):
 
   return cfg;
 }
+
 /**
  * Load the settings file or create it if not present.  If the settings have
  * already been loaded, return it without re-loading from disk.
  */
 export function load(): Settings {
+  let deploymentProfiles: DeploymentProfileType = { defaults: {}, locked: {} };
+  let setDefaultMemory = false;
+  let lockedProfileSettings: RecursivePartial<Settings>;
+
+  try {
+    deploymentProfiles = readDeploymentProfiles();
+    lockedProfileSettings = deploymentProfiles.locked;
+  } catch {
+    lockedProfileSettings = {};
+  }
   try {
     settings ??= loadFromDisk();
   } catch (err: any) {
     settings = clone(defaultSettings);
     if (err.code === 'ENOENT') {
-      _isFirstRun = true;
-      if (os.platform() === 'darwin' || os.platform() === 'linux') {
-        const totalMemoryInGB = os.totalmem() / 2 ** 30;
-
-        // 25% of available ram up to a maximum of 6gb
-        settings.virtualMachine.memoryInGB = Math.min(6, Math.round(totalMemoryInGB / 4.0));
+      if (Object.keys(deploymentProfiles.defaults).length) {
+        _.merge(settings, deploymentProfiles.defaults);
+        if (!_.has(deploymentProfiles.defaults, 'virtualMachine.memoryInGB')) {
+          setDefaultMemory = true;
+        }
+      } else {
+        _isFirstRun = true;
+        setDefaultMemory = true;
       }
     }
+    if (setDefaultMemory && (os.platform() === 'darwin' || os.platform() === 'linux')) {
+      const totalMemoryInGB = os.totalmem() / 2 ** 30;
+
+      // 25% of available ram up to a maximum of 6gb
+      settings.virtualMachine.memoryInGB = Math.min(6, Math.round(totalMemoryInGB / 4.0));
+    }
     if (os.platform() === 'linux' && !process.env['APPIMAGE']) {
       settings.application.updater.enabled = false;
+    } else {
+      const appVersion = getProductionVersion();
+
+      // Auto-update doesn't work for CI or local builds, so don't enable it by default
+      if (appVersion.includes('-') || appVersion.includes('?')) {
+        settings.application.updater.enabled = false;
+      }
     }
+  }
+  _.merge(settings, lockedProfileSettings);
```

Comment:
This is functionally equivalent to
```suggestion
  _.merge(settings, deploymentProfiles.locked);
```
(If `readDeploymentProfiles` threw, the assignment never took place, so `.locked` is already set to `{}`.)

Reply:
removed